saleor/webhook/transport/synchronous/transport.py

```diff
@@ -308,6 +308,13 @@ def trigger_webhook_sync(
     pregenerated_subscription_payload: dict | None = None,
 ) -> dict[Any, Any] | None:
     """Send a synchronous webhook request."""
+    if transaction.get_connection().in_atomic_block:
+        # Sync webhooks should be delivered after the transaction is committed.
+        # Otherwise the delivery task may not be able to fetch all the required data and
+        # the delivery may fail.
+        logger.error("Sync webhook was triggered inside a transaction: %s", event_type)
```

Comment:
`logger.exception` should provide us more information about the places where this logic is called.